src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1585,7 +1597,134 @@ else if (ticket == TicketType.POST_TELEPORT) // post teleport
                 cc.s.getLevel().getChunkSource().addRegionTicket(TicketType.UNKNOWN, target, radius, target);
             return new NumericValue(ticket.timeout());
         });
+        // =======================================
+        expression.addContextFunction("save_structure_template", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerLevel world = cc.s.getLevel();
+
+            StructureManager structureManager = world.getStructureManager();
+            StructureTemplate theStructure;
+            String name;
+
+            boolean returnnbt = lv.get(0).isNull();
+            ResourceLocation struident = null;
+            if (!returnnbt) {
+                name = lv.get(0).getString();
+                struident = ResourceLocation.tryParse(name);
+                if (name.isEmpty() || name.endsWith(":") || struident == null) {
+                    return Value.NULL;
+                } // if(lv.size()==2&&lv.get(1).isNull()){lv3.remove(struident);return
+                  // Value.NULL;}
+                  theStructure = structureManager.getOrCreate(struident);
+            } else {
+                theStructure = new StructureTemplate();
+            }
+            BlockArgument start = BlockArgument.findIn((CarpetContext) c, lv, 1);
+            BlockArgument end = BlockArgument.findIn((CarpetContext) c, lv, start.offset);
+            BlockArgument ignoredblock = BlockArgument.findIn((CarpetContext) c, lv, end.offset + 1, true, true, false);
+
+            BlockPos startBlockPos = new BlockPos(
+                    Math.min(start.block.getPos().getX(), end.block.getPos().getX()),
+                    Math.min(start.block.getPos().getY(), end.block.getPos().getY()),
+                    Math.min(start.block.getPos().getZ(), end.block.getPos().getZ()));
+            BlockPos endBlockPos = new BlockPos(
+                    Math.max(start.block.getPos().getX(), end.block.getPos().getX()),
+                    Math.max(start.block.getPos().getY(), end.block.getPos().getY()),
+                    Math.max(start.block.getPos().getZ(), end.block.getPos().getZ()));
+            BlockPos size = endBlockPos.subtract(startBlockPos).offset(1, 1, 1);
+            // name,start,end, ignoreEntities, ignoredBlock,~~author~~,disk
+            theStructure.fillFromWorld(world,
+                    startBlockPos,
+                    size,
+                    !lv.get(end.offset).getBoolean(),
+                    ignoredblock.block == null ? null : ignoredblock.block.getBlockState().getBlock());
+            // lv6.setAuthor(lv.get(9).getString());//MC-140821
+            if (!returnnbt) {
+                if (lv.get(ignoredblock.offset).getBoolean()) {
```

Comment:
same for saving parameter - honestly why it even is there? Why would you want to fill from world, and not save it? To save on write time? then you should use the structure and pass it to the variable?

Please give a usecase where save_structure('name', .... false) would ever make any sense?

(relying on structure manager to hold structures makes 0 sense to me by the way - you get variables for it).

Reply:
> relying on structure manager to hold structures makes 0 sense to me


Because I don't think it makes sense to serialize to/deserialize from a nbt value every time you load/unload. And it might be necessary to load content that the user saved manually with a structure block.